Dockerfile

```diff
@@ -1,62 +1,56 @@
-FROM python:3.10-slim
+FROM nvidia/cuda:12.0.1-base-ubuntu22.04
 
 # Set environment variables
-ENV PYTHONDONTWRITEBYTECODE=1 \
-    PYTHONUNBUFFERED=1 \
-    DEBIAN_FRONTEND=noninteractive \
-    PIP_NO_CACHE_DIR=1 \
-    CONFIG_PATH=/config/config.json
-
-# Create app user
-RUN groupadd -r squishy && \
-    useradd -r -g squishy -d /app -s /bin/bash squishy
-
-# Install system dependencies
-RUN apt-get update && \
-    apt-get install -y --no-install-recommends \
+ENV DEBIAN_FRONTEND=noninteractive
+ENV PYTHONUNBUFFERED=1
+ENV PYTHONDONTWRITEBYTECODE=1
+
+# Install system dependencies including ffmpeg and CUDA development libraries
+RUN apt-get update && apt-get install -y \
+    python3 \
+    python3-pip \
+    python3-dev \
     ffmpeg \
```

Comment:
The dependencies for VAAPI must remain in the Dockerfile, alongside the CUDA dependencies, so that we can ensure proper support for both types of hardware acceleration.